utils/check/handle_rpms.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+set -uex
+
+# This file does an install of the RPMs found in "${REPO_DIR}" and then
+# finds all the packages that the binaries installed reference.
+
+# The file ${PACKAGES} will contain a list of packages that the RPMs depend
+# on.
+
+# The file ${SOFILES} will contain a list of sharable images depended on by
+# the RPMs.  The default name for ${SOFILES} is "daos_depends_libries".
+
+# This will likely need to be edited for other build systems.
+
+: "${WORKSPACE:="${PWD}"}"
+
+: "${RPM_DIR:="${WORKSPACE}/rpm_dir"}"
+
+: "${SOFILES:="daos_depends_libraries"}"
+: "${PACKAGES:="daos_depends_packages"}"
+
+
+repo_file="/etc/yum.repos.d/"
+repo_file+="repo.dc.hpdd.intel.com_repository_daos-stack-"
+repo_file+="el-7-x86_64-stable-local.repo"
+
+repo_url="https://repo.dc.hpdd.intel.com/repository/"
+repo_url+="daos-stack-el-7-x86_64-stable-local"
+
+if [ ! -e "${repo_file}" ]; then
+  cat << EOF > daos_repo_file
+
+[repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local]
+name=added from: ${repo_url}
+baseurl=${repo_url}
+enabled=1
+
+gpgcheck = False
+
+EOF
+  sudo cp daos_repo_file "${repo_file}"
+  sudo yum -y makecache
+fi
+
+sudo yum -y remove cart cart-debuginfo cart-devel cart-tests \
+                   daos daos-client daos-debuginfo daos-devel daos-server \
+                   daos-tests CUnit dpdk fio fuse fuse-lib hwloc \
+                   libabt0 libcmocka libfabric libfabric-devel libisa-l \
+                   libopa1 libpmem libpmemblk mercury ndctl\
+                   ompi openmpi3 openpa-devel \
+                   pexpect protobuf-c pmix \
+                   python-configshell python-urwid \
+                   raft raft-debuginfo raft-devel \
+                   spdk spdk-tools
+
+sudo yum -y install "${RPM_DIR}"/*.rpm
+
+set +x
+
+# Step 1. Get a list of all the files installed by the RPMs that
+#         we just installed.
+f_list=""
+
+for f in "${RPM_DIR}"/*.rpm; do
+  rpm_files="$(rpm -qlp "${f}")"
+  f_list+=" ${rpm_files}"
+done
+
+# Step 2. Identify which of the files are ELF executables and
+#         lookup what images that they depend on.
+rm -f raw_image_depends
+
+for f in ${f_list}; do
+  if [[ -x "${f}" ]]; then
+    if file "${f}" | grep ": ELF "; then
+      ldd "${f}" >> raw_lib_depends
+    fi
+  fi
+done
+
+# Step 3. Strip out just the filenames, sort removing duplicates
+awk '{print $1}' raw_lib_depends | sort -u > "${SOFILES}"
+
+# Step 4. Find the package that provided the dependent file.
+rm -rf raw_pkg_depends
+
+while read dp; do
+  if [ "${dp}" != linux-vdso.so.1 ]; then
+    if [[ ${dp} = /* ]]; then
+      #echo "Testing ${dp}"
+      rpm -qf "/usr/${dp}" >> raw_pkg_depends
+    else
+      #echo "testing ${dp}"
+      dp_found="$(sudo find /usr -name "${dp}")"
+      for dpf in ${dp_found}; do
+        rpm -qf "${dpf}" >> raw_pkg_depends
+      done
+    fi
+  fi
+done < "${SOFILES}"
+
+sort -u raw_pkg_depends > "${PACKAGES}"
```

Comment:
Perhaps I am not fully understanding what the above is supposed to do, but I think a much quicker, more concise way of achieving the above is:

```bash
$ repoquery -q --requires daos{,-{client,server}} | xargs repoquery -q --whatprovides | sort -u
```

Maybe compare your `raw_pkg_depends` to the output of the above and see if they are for all intents and purposes, the same.

Reply:
This script is to query what the binaries are using, not what the information in the rpm says that they should be using.

The above query produces multiple packages that could provide the packages, not the packages that are actually installed, which is what is needed for the task.

Reply:
RPMs also assume a base set of packages are installed with the distro, and those packages do not need to be listed in the Requires: in the RPMs.  I need those packages also.

The requirement for this script was to find the installed version of every specific package that supplies a library used the specific set of  RPMs that were built in CI.

Reply:
> RPMs also assume a base set of packages are installed with the distro,

This is actually not true.  A properly written RPM should depend on **everything** it needs such that you could install just it into an empty chroot and everything needed for it to run is installed.  This is mostly achieved through [Automatic Dependencies](http://ftp.rpm.org/max-rpm/s1-rpm-depend-auto-depend.html) although of course not *everything* is automatically determinable, hence the need for manual `Requires:`.  But those manual `Requires:` should be fully complete and `mock` helps to ensure that.

To be clear, of course an RPM doesn't need to `Requires:` (automatically or manually) every RPM it might need.  That is the job of the RPMs it directly depends on, and the RPMs that they depend on etc.  But ultimately from a given RPM you should be able to draw a graph to every single other RPM needed for it to fully function.